File: src/main/GO/leetcode/lists/topinterview/LeetCode149.go
package topinterview

func maxPoints(points [][]int) int {
	n := len(points)
	if n <= 2 { //fix
		return n
	}
	res := 2
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			//fix
			count := 2
			for k := j + 1; k < n; k++ {
				if (points[i][0]-points[j][0])*(points[j][1]-points[k][1]) ==
					(points[i][1]-points[j][1])*(points[j][0]-points[k][0]) {
					count++
				}
			}
			res = Max(res, count)
		}
	}
	return res
}

File: java/src/leetcode/problems/LeetCode149.java
package leetcode.problems;

/**
 * @author aiwujingxin@gmail.com
 * @date 2022/7/11 23:32
 */
public class LeetCode149 {

    public int maxPoints(int[][] points) {
        int n = points.length;
        if (n <= 2) {
            return n;
        }
        int res = 2;
        for (int i = 0; i < n - 2; i++) {
            for (int j = i + 1; j < n - 1; j++) {
                int x = points[j][0] - points[i][0];
                int y = points[j][1] - points[i][1];
                int count = 2;
                for (int k = j + 1; k < n; k++) {
                    int x1 = points[k][0] - points[j][0];
                    int y1 = points[k][1] - points[j][1];
                    // x/y = x1/y1
                    if (x * y1 == y * x1) {
                        count++;
                    }
                }
                res = Math.max(res, count);
            }
        }
        return res;
    }
}
